api/client/commands.go

```diff
@@ -1674,60 +1697,86 @@ func (cli *DockerCli) CmdPs(args ...string) error {
 		return ss
 	}
 
-	for _, out := range outs.Data {
-		outID := out.Get("Id")
-
-		if !*noTrunc {
-			outID = common.TruncateID(outID)
-		}
-
-		if *quiet {
-			fmt.Fprintln(w, outID)
-
-			continue
-		}
-
-		var (
-			outNames   = stripNamePrefix(out.GetList("Names"))
-			outCommand = strconv.Quote(out.Get("Command"))
-			ports      = engine.NewTable("", 0)
-		)
+	type containerMeta struct {
+		c string
+		i string
+		m string
+		t string
+		s string
+		p string
+		n string
+		z string
+	}
 
+	outp := make([]containerMeta, 0)
+	for _, out := range outs.Data {
+		portTable := engine.NewTable("", 0)
```

Comment:
I'm not sure I know exactly what engine.Table stuff really does. Any pointers on what other commands are doing that depend on the table stuff? I could come up with a strawman but I don't know what y'all want.